apps/06.1Checkers.js

```diff
@@ -3,75 +3,199 @@
+    // creates an 8x8 array, filled with null values
+    this.createGrid = function() {
+        // loop to create the 8 rows
+        for (var row = 0; row < 8; row++) {
+            this.grid[row] = [];
+            // push in 8 columns of nulls
+            for (var column = 0; column < 8; column++) {
+                this.grid[row].push(null);
+            }
+        }
+    };
+
+    // prints out the board
+    this.viewGrid = function() {
+        // add our column numbers
+        var string = "  0 1 2 3 4 5 6 7\n";
+        for (var row = 0; row < 8; row++) {
+            // we start with our row number in our array
+            var rowOfCheckers = [row];
+            // a loop within a loop
+            for (var column = 0; column < 8; column++) {
+                // if the location is "truthy" (contains a checker piece, in this case)
+                if (this.grid[row][column]) {
+                    // push the symbol of the check in that location into the array
+                    rowOfCheckers.push(this.grid[row][column].symbol);
+                } else {
+                    // just push in a blank space
+                    rowOfCheckers.push(' ');
+                }
+            }
+            // join the rowOfCheckers array to a string, separated by a space
+            string += rowOfCheckers.join(' ');
+            // add a 'new line'
+            string += "\n";
+        }
+        console.log(string);
+    };
+    this.selectChecker = function(row, column) {
+        return this.grid[row][column];
+
+    };
+    this.killChecker = function(position) {
+
+        var checker = this.selectChecker(position[0], position[1]);
 
-  // Your code here
+        var name = this.checkers.indexOf(checker);
+
+        this.checkers.splice(name, 1);
+
+        this.grid[position[0]][position[1]] = null;
+        //In your Board class, write a method killChecker that take a single argument position which is a coordinate pair, eg. [0, 5].
+        //In it, use this.selectChecker to grab the checker at the position given.
+        //Find the index of that checker in the this.checkers array. then remove it by .splice()ing it out.
+        //Then assign the position on this.grid to null. That checker is dead.
+
+    };
+    // Your code here
 }
+
 function Game() {
 
-  this.board = new Board();
+    this.board = new Board();
+
+    this.start = function() {
+        this.board.createGrid();
+        // Your code here
+        this.board.createCheckers();
+        //start might look like this 40, end might look like this 52.
+    };
+    this.moveChecker = function(start, end) {
+        //we need to send in row and col to select checker
+        //"4" "0"
+        var startrow = start.split('')[0];
+        var startcol = start.split('')[1];
+        var endrow = end.split('')[0];
+        var endcol = end.split('')[1];
+        var sum = startrow + endrow;
+        var checker = this.board.selectChecker(startrow, startcol, endrow, endcol);
```

Comment:
You pass in 4 arguments into your `this.board.selectChecker()` function when on line 133 you only establish two parameters for this function

JavaScript is kind enough to completely bypass your 3rd & 4th argument here but make sure your arguments passed and function parameters are always in sync.